### PR TITLE
adds aggregated_by method to BaseExtension

### PR DESCRIPTION
**Pull request by flyingzumwalt**
This is ready for review now @jcoyne @terrellt

### CODE REVIEW

File: lib/active_fedora/aggregation/base_extension.rb [hunk 1]
```diff
@@ -3,6 +3,13 @@ module BaseExtension
     extend ActiveSupport::Concern
     include PersistLinks
 
+    # Queries the RDF graph to find all aggregations that include this object
+    def aggregated_by
+      proxy_uris = self.resource.query(predicate: RDF::Vocab::ORE.proxyFor, object: RDF::URI(self.uri)).subjects.to_a
```
Comment (flyingzumwalt) [inline lib/active_fedora/aggregation/base_extension.rb]: This is the line where it should be finding the proxies but isn't finding anything.

**Comment by flyingzumwalt** [comment]
This blocks projecthydra-labs/hydra-pcdm#114

File: lib/active_fedora/aggregation/base_extension.rb [hunk 2]
```diff
@@ -3,6 +3,16 @@ module BaseExtension
...
     include PersistLinks
 
+    # Queries the RDF graph to find all aggregations that include this object
+    def aggregated_by
+      # In theory you should be able to find the aggregation predicate (ie ore:aggregates)
+      # but Fedora does not return that predicate due to this bug in FCREPO: https://jira.duraspace.org/browse/FCREPO-1497 
+      # so we have to, instead, look up the proxies asserting RDF::Vocab::ORE.proxyFor and return their containers.
+      proxy_uris = self.resource.query(predicate: RDF::Vocab::ORE.proxyFor, object: RDF::URI(self.uri)).subjects.to_a
+      proxy_objects = proxy_uris.map{|proxy_uri| ActiveFedora::Aggregation::Proxy.find(ActiveFedora::Base.uri_to_id(proxy_uri.to_s))}
+      proxy_objects.map(&:container) # This currently doesn't find anything because proxyIn (which Proxy#containers uses) isn't asserted in the graph
```
Comment (terrellt) [inline lib/active_fedora/aggregation/base_extension.rb]: This comment shouldn't be true anymore, due to #39

File: lib/active_fedora/aggregation/base_extension.rb [hunk 3]
```diff
@@ -3,6 +3,16 @@ module BaseExtension
     extend ActiveSupport::Concern
     include PersistLinks
 
+    # Queries the RDF graph to find all aggregations that include this object
+    def aggregated_by
```
Comment (terrellt) [inline lib/active_fedora/aggregation/base_extension.rb]: Can you add a @return YARD comment?

File: lib/active_fedora/aggregation/base_extension.rb [hunk 4]
```diff
@@ -3,6 +3,16 @@ module BaseExtension
...
+      # In theory you should be able to find the aggregation predicate (ie ore:aggregates)
+      # but Fedora does not return that predicate due to this bug in FCREPO: https://jira.duraspace.org/browse/FCREPO-1497 
+      # so we have to, instead, look up the proxies asserting RDF::Vocab::ORE.proxyFor and return their containers.
+      proxy_uris = self.resource.query(predicate: RDF::Vocab::ORE.proxyFor, object: RDF::URI(self.uri)).subjects.to_a
```
Comment (terrellt) [inline lib/active_fedora/aggregation/base_extension.rb]: Can you do `object: rdf_subject` instead?

File: spec/activefedora/association_spec.rb [hunk 5]
```diff
@@ -105,6 +105,25 @@ class Image < ActiveFedora::Base
...
+          image2.save
+        end
+        subject { generic_file1.reload }
+        it "can find the record that contains them" do
+          expect(subject.aggregated_by).to eq [image]
+        end
+        context "when multiple records contain an element" do
+          subject { generic_file2.reload }
+          it "can find all of the records that contain it" do
+            expect(subject.aggregated_by).to include(image2,image)
```
Comment (terrellt) [inline spec/activefedora/association_spec.rb]: Can you change include to `contain_exactly` ?

File: lib/active_fedora/aggregation/base_extension.rb (hunk 4 above)
Comment (flyingzumwalt) [inline lib/active_fedora/aggregation/base_extension.rb]: I don't understand what you mean by that.
This does not work: 
`proxy_uris = self.resource.query(predicate: RDF::Vocab::ORE.proxyFor, object: rdf_subject)`

Comment (terrellt) [inline lib/active_fedora/aggregation/base_extension.rb]: Really? I would have expected RDF::URI(self.uri) and self.rdf_subject to be the same thing.

File: spec/activefedora/association_spec.rb [hunk 6]
```diff
@@ -105,6 +105,24 @@ class Image < ActiveFedora::Base
           expect(subject).to eq image.uri
         end
       end
+      
+      describe "its elements" do
```
Comment (terrellt) [inline spec/activefedora/association_spec.rb]: Aren't you describing #aggregated_by ?

File: lib/active_fedora/aggregation/base_extension.rb (hunk 4 above)
Comment (flyingzumwalt) [inline lib/active_fedora/aggregation/base_extension.rb]: I don't understand what you are suggesting.  If you want to offer a replacement for this line, I will try it.

Comment (terrellt) [inline lib/active_fedora/aggregation/base_extension.rb]: proxy_uris = self.resource.query(predicate: RDF::Vocab::ORE.proxyFor, object: rdf_subject).subjects.to_a

**Pull request merged**